packages/fx-core/src/plugins/solution/fx-solution/commonQuestions.ts

```diff
@@ -77,8 +77,34 @@ export async function checkSubscription(
       )
     );
   }
-  const askSubRes = await ctx.azureAccountProvider!.getSelectedSubscription(true);
-  return ok(askSubRes!);
+
+  const subscriptionId = ctx.envInfo.config.azure.subscriptionId;
+  if (!isMultiEnvEnabled() || !subscriptionId) {
+    const askSubRes = await ctx.azureAccountProvider.getSelectedSubscription(true);
+    return ok(askSubRes!);
+  }
+
+  // make sure the user is logged in
+  await ctx.azureAccountProvider.getAccountCredentialAsync(true);
```

Comment:
Do we need to check if user cancel the login dialog?

Reply:
The current behavior of `azureAccountProvider` is throwing exceptions on cancel or login failure, and core will handle exceptions. So the current implemention will work on these cases.

Although there is `undefined` in the method signature, there is no such cases when `undefined` is returned. Need to confirm with @xiaolang124.